rpc/server.go

```diff
@@ -112,7 +149,31 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	security.LogTLSState("RPC", r.TLS)
 	io.WriteString(conn, "HTTP/1.0 "+connected+"\n\n")
-	s.serveConn(conn, authHook)
+
+	codec := codec.NewServerCodec(conn, authHook)
+	responses := make(chan serverResponse)
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		s.readRequests(codec, responses)
+		wg.Done()
+	}()
+	go func() {
+		s.sendResponses(codec, responses)
+		wg.Done()
+	}()
+	wg.Wait()
```

Comment:
Not that it matters, but I think you could use the current goroutine for either reading requests or sending responses.